x-pack/plugins/search_indices/public/components/start/hooks/use_indices_redirect.tsx

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect } from 'react';
+
+import type { IndicesStatusResponse } from '../../../../common';
+
+import { useKibana } from '../../../hooks/use_kibana';
+
+import { navigateToIndexDetails } from './utils';
+
+export const useIndicesRedirect = (indicesStatus?: IndicesStatusResponse) => {
+  const { application, http } = useKibana().services;
+  return useEffect(() => {
+    if (!indicesStatus) return;
+    if (indicesStatus.indexNames.length === 0) return;
+    if (indicesStatus.indexNames.length === 1) {
+      navigateToIndexDetails(application, http, indicesStatus.indexNames[0]);
+      return;
+    }
+    application.navigateToApp('management:index_management');
+  }, [application, http, indicesStatus]);
```

Comment:
suggestion: Im thinking of the issues where the indices created isn't just one. Maybe the conditional here needs to be greater than 1 and we redirect them to the last index.

Reply:
That was my original intention, the problem is with serverless there is no way for us to know what the "last" index is. We'll be reliant on the sort order from the query which I don't believe will be related to creation at all. And there are not creation dates available in the settings in serverless that I could find.

Reply:
yeh thats a good point, theres no concept of creation time for indices. Should we change the conditional to be greater than 1 though as user wont be directed if created more than one?

Reply:
If there is > 1 index they are redirected to the index list page.

Reply:
OK great! i missed that code :)

Reply:
also of note, this logic is also used on the first page load not just for polling. So if you open the start page and you have 10 indices this code is run.

So the logic is:

0 indices - render the page
1 index - redirect to that index's details page
> 1 index - redirect to index management index list page

Reply:
Yeh makes sense, seen the last line now. I tried locally (adding multiple indices) and was redirected to 404 on `app/management:index_management` BTW

Reply:
hmm, will test again and fix that

Reply:
updated to `application.navigateToApp('management', { deepLinkId: 'index_management' });` and tested that it worked as expected. I'll see about making an FTR test for this too.